### PR TITLE
ENS Docs

### PR DESCRIPTION
**Pull request by eggplantzzz**
This PR provides documentation of the ENS integration.

### CODE REVIEW

File: src/docs/truffle/advanced/ethereum-name-service.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+As part of this ENS integration, a new `ens` module is available on the
+deployer object available during migration functions. Currently there is a
+single method on this module named `setAddress`. If you own a domain name,
+you can use this method to deploy a resolver and set its address. If the
+resolver already exists, it will set the address that the resolver references
+if it is not the same as the input address.
+
+The signature for this method is
+`setAddress(<name: string>, <addressOrContract: string|truffle contract instance>, <from: string)`.
```
Comment (gnidan) [inline src/docs/truffle/advanced/ethereum-name-service.md]: Typo

```suggestion
`setAddress(<name: string>, <addressOrContract: string|truffle contract instance>, <from: string>)`.
```

Or maybe just skip the <>s:
```suggestion
`setAddress(name: string, addressOrContract: string | TruffleContractInstance>, from: string)`.
```

File: src/docs/truffle/advanced/ethereum-name-service.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  - The `from` parameter is the address that you want to send the transaction
+  from. This must be the address that owns the domain in question. If this
+  address does not own the domain then the transaction will fail.
+
+## Automatic registry deployment
+
+Another feature of this integration is the ability to deploy an ENS registry on
+a network. This would be useful when developing locally and testing. If
+Truffle cannot connect to an ENS registry on the network you are using, it
+will check to see if you have "dev mode" enabled. If you do, it will attempt
```
Comment (gnidan) [inline src/docs/truffle/advanced/ethereum-name-service.md]: Maybe take out the note about "dev mode" because you don't explain it anymore.

**Comment by adrianmcli** [comment]
@eggplantzzz Sorry, the capitalization was bothering me lol.

File: src/docs/truffle/advanced/ethereum-name-service.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+```javascript
+module.exports = {
+  ens: {
+    enabled: true
+  }
+}
+```
+
+With the above config, you would be able to resolve names on Mainnet, Ropsten,
+Rinkeby, and Goerli. Truffle connects to the "official" ENS-deployed registries
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply src/docs/truffle/advanced/ethereum-name-service.md]: Sounds good, I agree

**Pull request merged**